halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -88,18 +88,112 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
 	}
 }
 
-func TestHappyPathDelivery(t *testing.T) {
+func TestDeliveryWithBrokenServer(t *testing.T) {
 	t.Parallel()
 
 	deliverInterval := int64(3)
 	ethStake := int64(7)
+	privKey := k1.GenPrivKey()
+
+	ethClientMock, err := ethclient.NewEngineMock(
+		ethclient.WithPortalRegister(netconf.SimnetNetwork()),
+		ethclient.WithMockValidatorCreation(privKey.PubKey()),
+		ethclient.WithMockSelfDelegation(privKey.PubKey(), ethStake),
+	)
+	require.NoError(t, err)
+	sServer := brokenMsgServerStub{errors.New("unconditional error")}
+
+	keeper, ctx := setupKeeper(t, deliverInterval, ethClientMock, new(authKeeperStub), new(bankKeeperStub), new(stakingKeeperStub), &sServer)
+
+	var hash common.Hash
+	events, err := keeper.Prepare(ctx, hash)
+	require.NoError(t, err)
+
+	expectDelegates := 1
+	expectCreates := 1
+	expectTotalEvents := expectDelegates + expectCreates
+
+	require.Len(t, events, expectTotalEvents)
+
+	for _, event := range events {
+		err := keeper.parseAndDeliver(ctx, &event)
+		require.True(t, strings.Contains(err.Error(), sServer.err.Error()))
```

Comment:
nit: Incorrect syntax usage of **_require_** , update all usages.

```suggestion
	require.Error(t, err)
	require.Contains(t, err.Error(),"create validator: pubkey to cosmos")
```
Using require.Error and require.Contains is preferred as it provides clearer intent, better failure messages, and avoids potential panics if err is nil. It also aligns with idiomatic testing practices for improved readability and debugging.

Reply:
Nice! Thanks!